plugins/bundle/plugin.go

```diff
@@ -429,3 +479,96 @@ func (p *Plugin) configDelta(newConfig *Config) (map[string]*Source, map[string]
 
 	return newBundles, updatedBundles, deletedBundles
 }
+
+func saveBundleToDisk(path string, name string, b *bundle.Bundle) error {
+
+	bundleDir := filepath.Join(path, name)
+	var backupBundleDir string
+
+	// if a bundle already exists, create a backup and then write the new bundle
```

Comment:
We might want to consider adjusting the order for these. Making the backup and restoring on error is good, but if OPA was stopped/crashed/etc mid-way through writing the new one I think we'd end up in trouble (the deferred calls to restore the backup would never get called)

A common strategy for this kind of thing is to write into a temporary file, and only after a successful write with the whole thing we swap it in with a like `mv` sort of operation on the filesystem so it is atomic. The goal being that you don't allow for a period of time where the file we've persisted is in a partially written state. From a high level I'd imagine the steps to be more like:

* Write to `filepath.Join(bundleDir, ".bundle.tar.gz.tmp")`. or whatever we want to call it, the gist being that its in the same directory and similar name but is a hidden file and somewhat obvious that its a partial or temp file.
* If it fails, delete that new/temporary file. No restore needed, the original is still there.
* If it succeeds, use `os.Rename()` to change the tmp file to the "real" path. This will overwrite the old version (same as like `mv -f`)

Reply:
Updated the backup/restore logic.